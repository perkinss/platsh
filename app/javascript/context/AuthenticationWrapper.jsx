import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
