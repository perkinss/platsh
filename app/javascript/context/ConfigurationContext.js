import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
