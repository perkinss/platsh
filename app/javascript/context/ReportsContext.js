import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
