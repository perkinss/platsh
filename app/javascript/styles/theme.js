import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
