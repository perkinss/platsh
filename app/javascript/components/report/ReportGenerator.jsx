import React, { useContext, useEffect, useState } from 'react'
import {Link as RouterLink} from 'react-router-dom'
import {
    Button,
    Checkbox,
    CircularProgress,
    Divider,
    Fade,
    FormControl,
    FormControlLabel,
    Grid,
    Input,
    InputLabel,
    Link,
    List,
    ListItem,
    ListItemIcon,
    ListSubheader,
    makeStyles,
    MenuItem,
    Paper,
    Select,
    Slide,
    SwipeableDrawer,
    Tooltip,
    Typography,
} from "@material-ui/core";
import BuildIcon from "@material-ui/icons/BuildRounded";
import {ReportsContext} from "../../context/ReportsContext";
import generateTopicText, { generateCompetencyText } from "./CommentGenerator";
import {fetchCompetencyScore, fetchHeatMap, fetchTopicMarks} from "../../api/reportApi";

const drawerWidth = '40%'

const useStyles = makeStyles((theme) => ({
    root: {
        '& .MuiTextField-root': {
            margin: theme.spacing(1),
            width: '25ch',
        },
    },
    drawer: {
        width: drawerWidth,
        flexShrink: 0,
    },
    drawerPaper: {
        width: drawerWidth,
    },
    list: {
        width: 250,
    },
    fullList: {
        width: 'auto',
    },
    listItem_4: {
        color: 'rgba(8,255,0,0.6)'
    },
    listItem_3: {
        color: 'rgba(255,242,0,0.8)'
    },
    listItem_2: {
        color: 'rgba(255,99,1,0.7)'
    },
    listItem_1: {
        color: 'rgba(255,0,0,0.48)'
    }
}));

const makeHeaderStyles =  makeStyles(theme => ({
    root: {
        background: theme.palette.primary.dark,
        opacity: '.8',
        fontSize: '2em',
        width: '100%'
    },

}))

export default function ReportGenerator (props) {

    const {showGenerator, student, closeGenerator, section, fromDate, toDate, selectedTopics } = props

    if (!showGenerator) {
        return null
    }

    const { state, dispatch } = useContext(ReportsContext);
    const [selectedHighCompetencies, setSelectedHighCompetencies] = useState([])
    const [selectedLowCompetencies, setSelectedLowCompetencies] = useState([])
    const [reportText, setReportText] = useState("")
    const [competencyReport, setCompetencyReport] = useState([])
    const [competencyDrawerOpen, setCompetencyDrawerOpen] = React.useState(false);

    const classes = useStyles()

    const headerClasses = makeHeaderStyles()
    const getMedian = (sortedNumbers) => {
        const middle = Math.floor(sortedNumbers.length / 2)
        const isEven = (sortedNumbers.length & 1) === 0

        return isEven ? ((sortedNumbers[middle] + sortedNumbers[middle + 1]) / 2) : sortedNumbers[middle]
    }

    const shouldFetch = (slice) => {
        return section && student && (!slice || !slice[section.id] || !slice[section.id][student.id])
    }

    /**
     * Returns true if the competency data (scores and heat map) for the selected student has been set in the state.
     */
    const competencyDataIsLoaded = () => {
        return (section && student
            && !shouldFetch(state.competency_averages)
            && !shouldFetch(state.heat_map)
        )
    }

    /**
     * This effect waits for the heat map and competency scores for the currently selected student to finish loading
     * into state, then runs the calculations for the median competency score and otherwise prepares the competency
     * selection drawer for the competency-based comments.
     */
    useEffect(() => {
        if (competencyDataIsLoaded()) {
            let student_competencies = state.heat_map[section.id][student.id]
            let modes = state.competency_averages[section.id][student.id].competency_report
                .filter(course => Object.keys(course.competency_modes).length > 0)
                .reduce((obj, course) =>{
                    return {...obj, [course.id]: course.competency_modes}
                },{})

            let competencyReport = Object.keys(modes).map(courseId => {
                let course = student_competencies.find(c => `${c.id}` === `${courseId}`)
                let course_modes = Object.keys(modes[courseId]).map(competencyId => {
                    let competency = course.data.find(c => `${c.id}` === `${competencyId}`)
                    let count = 0
                    if (competency) {
                        count = competency.data.reduce((total, number) => {
                            return total ? total + number.count : number.count
                        }, 0)
                    }
                    return {id: competencyId, name: competency.name, phrasing: competency.phrasing, score: modes[courseId][competencyId], count: count}
                })
                course_modes.sort((a, b) => {
                    return b.score - a.score || b.count - a.count
                })
                let median = getMedian(course_modes.map(a => a.score))
                let highCompetencies = course_modes.filter(m => m.score >= median)
                let lowCompetencies = course_modes.filter(m => m.score < median)
                return {
                    id: course.id,
                    title: course.title,
                    competencies: course_modes,
                    highCompetencies: highCompetencies,
                    lowCompetencies: lowCompetencies,
                    median: median
                }
            })
            setCompetencyReport(competencyReport)
        }
    }, [competencyDataIsLoaded()])

    /**
     * This effect loads the topic averages and competency-related data after the student has been selected or re-selected:
     */
    useEffect( () => {
        if (student) {
            setReportText('')
            if (shouldFetch(state.topic_averages)) {
                fetchTopicMarks(dispatch, section, student, fromDate, toDate)
            }
            if (shouldFetch(state.competency_averages)) {
                fetchCompetencyScore(dispatch, section, student, fromDate, toDate)
            }
            if (shouldFetch(state.heat_map)) {
                setSelectedHighCompetencies([])
                setSelectedLowCompetencies([])
                fetchHeatMap(dispatch, section, student, fromDate, toDate)
            }
        }
    },[student])

    const handleSelectHighCompetency = (event) => {
        let copy = [...selectedHighCompetencies]

        if (event.target.checked) {
            // add to list
            setSelectedHighCompetencies([...selectedHighCompetencies, event.target.value])
        } else {
            // remove from list
            copy.splice(copy.indexOf(event.target.value), 1)
            setSelectedHighCompetencies(copy)
        }
    }

    const handleSelectLowCompetency = (event) => {
        let copy = [...selectedLowCompetencies]

        if (event.target.checked) {
            // add to list
            setSelectedLowCompetencies([...selectedLowCompetencies, event.target.value])
        } else {
            // remove from list
            copy.splice(copy.indexOf(event.target.value), 1)
            setSelectedLowCompetencies(copy)
        }
    }

    const handleReportChange = (event, id) => {
        setReportText(event.target.value);
    }

    const handleGenerateTopicComments = () => {
        const remarks = generateTopicText(state.topic_averages[section.id][student.id], selectedTopics.map(s => s.id), student)
        setReportText(reportText + " \n" + remarks)
    }

    const handleGenerateCompetencyComments = () => {
        const comments = generateCompetencyText(competencyReport, selectedHighCompetencies, selectedLowCompetencies, student)
        setReportText(reportText + " \n" + comments)
    }

    const toggleDrawer = (open) => (event) => {
        if (event && event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
            return;
        }
        setCompetencyDrawerOpen(open)
    }

    // Rendering the competency drawer -- get the list of competencyReport:
    function getCompetencyDrawerList(courseCompetencyData, onChangeHandler, selectedOnes) {
        return courseCompetencyData.map((competency, index) => {
            let score = competency.score
            const className = score > 3 ? classes.listItem_4 : score > 2 ? classes.listItem_3 : score > 1 ? classes.listItem_2 : classes.listItem_1

            const label = `${competency.name} ${ '(mode: ' + competency.score + ', based on ' + competency.count + ' observations)'}`
            return (
                <ListItem key={`${competency.id}-${index}`} button onChange={onChangeHandler} >
                    <ListItemIcon>
                        <Tooltip title={label} placement={"bottom-start"}>
                            <FormControlLabel
                                control={
                                    <Checkbox
                                        value={competency.id}
                                        checked={selectedOnes.includes(`${competency.id}`) || false}
                                    />}
                                label={label}
                                className={className}
                            />
                        </Tooltip>
                    </ListItemIcon>
                </ListItem>
            )
        });
    }

    // render the competency selection drawer:
    const competencyDrawer = () => {
        if (competencyDrawerOpen && competencyReport) {
            return (
                competencyReport.map( (course, index) => {
                    return (
                        <List
                            key={`competencies-${index}`}
                            component="div"
                            aria-labelledby="nested-list-subheader"
                            subheader={
                                <ListSubheader
                                    component="div"
                                    inset={true}
                                    classes={headerClasses}
                                    className={headerClasses.root}
                                    styles={{opacity: 1}}
                                    id="nested-list-subheader">
                                    {course.title}
                                </ListSubheader>
                            }
                            className={classes.root}
                        >
                            <div style={{ width: '100%', overflow: "hidden", textOverflow: "ellipsis" }}>

                                <Typography color={"secondary"} style={{paddingTop: '10px', paddingLeft: '15px'}}>
                                    High Scores &gt;= median {course.median}
                                </Typography>
                                { getCompetencyDrawerList(course.highCompetencies, handleSelectHighCompetency, selectedHighCompetencies) }
                            </div>
                            <Divider />
                            <div style={{ width: '100%', overflow: "hidden", textOverflow: "ellipsis" }}>
                                <Typography color={"secondary"} style={{paddingTop: '10px', paddingLeft: '15px'}}>
                                    Low Scores &lt; median {course.median}
                                </Typography>
                                { getCompetencyDrawerList(course.lowCompetencies, handleSelectLowCompetency, selectedLowCompetencies) }
                            </div>
                        </List>
                    )
                })
            )
        }

    }

    const getTopicSelectItem = (topic) => {
        const selectItem = <Typography aria-label={'topic name'}>{topic.name}</Typography>
        if (!topic.description) {
            return selectItem
        }
        return (
            <Tooltip
                TransitionComponent={Fade}
                TransitionProps={{ timeout: 600 }}
                placement={'top'}
                title={topic.description}
            >
                {<Typography aria-label={'topic name'}>{topic.name}</Typography>}
            </Tooltip>
        )
    }

    return (
        <>
            <Slide in={showGenerator} direction={"down"}>
                <div id={'student-report-generator'}>
                    <Grid container spacing={2} alignContent={'flex-end'} justify={'flex-start'}>
                        <Grid item xs={1} xl={1}>
                            <Button onClick={closeGenerator}>
                                <Typography variant={'caption'}>&lt; Back</Typography>
                            </Button>
                        </Grid>
                        <Grid item xs={10} xl={10}>
                            {student && <Typography variant={"h4"}>Comments for {student.name}</Typography>}
                            <Typography variant={"h5"}>{section.name}</Typography>
                        </Grid>
                        <Grid item xs={1} xl={1}>
                            <Button onClick={closeGenerator}>
                                <Typography variant={'h6'} onClick={closeGenerator}>x</Typography>
                            </Button>
                        </Grid>
                        {selectedTopics?.length > 0 &&
                        <Grid item sm={12} xl={12}>
                            <Typography align={"left"} variant={"h5"}>
                                Topics for the selected Reporting Period:
                            </Typography>
                            <br/>
                            <Grid
                                container
                                spacing={4}
                                alignContent={'center'}
                                justify={'flex-start'}
                                aria-label={'List of topics for selected reporting configuration'}
                            >
                                {selectedTopics.map((topic) => {
                                    return (
                                        <Grid key={`topic-${topic.id}`} item md={2} xs={12}>
                                            {getTopicSelectItem(topic)}
                                        </Grid>
                                    )
                                })
                                }
                            </Grid>
                            <br/>
                            {!student && <Typography variant={"body1"} align={"left"}>(Select a Student above to continue)</Typography>}
                        </Grid>
                        }
                        {student && competencyDataIsLoaded() &&
                        <>
                            <Grid item xs={6} xl={3}>
                                <div >
                                    <Button
                                        variant={'contained'}
                                        name={'competencyReport'}
                                        onClick={toggleDrawer(true)}
                                    >
                                        Select Competencies
                                    </Button>
                                    <SwipeableDrawer
                                        anchor={'right'}
                                        open={competencyDrawerOpen}
                                        onClose={toggleDrawer(false)}
                                        onOpen={toggleDrawer(true)}
                                        classes={{
                                            paper: classes.drawerPaper
                                        }}
                                        className={classes.drawer}
                                    >
                                        {competencyDrawer()}
                                    </SwipeableDrawer>
                                </div>
                            </Grid>
                            <Grid item xs={12} xl={12}>
                                <Paper className={classes.paper} elevation={10} style={{padding: '10px'}}>
                                    <Grid container spacing={1} alignContent={'flex-end'} justify={'space-between'}>
                                        {!shouldFetch(state.topic_averages) && <Grid item xs={12} xl={12}>
                                            {state.topic_averages[section.id][student.id].map(course => {
                                                let courseTopics = course.course_marks
                                                    .filter(mark => selectedTopics?.includes(`${mark.id}`))
                                                    .map(mark => mark.name)
                                                    .reduce((sentence, mark) => {
                                                        return sentence ? sentence + ", " + mark : mark
                                                    }, "")
                                                if (courseTopics && courseTopics.length > 0) {
                                                    return (
                                                        <Grid container spacing={1} alignContent={'flex-start'} justify={'space-between'}
                                                              key={`c-${course.id}`}>
                                                            <Grid item xs={4} xl={2}>
                                                                <Typography
                                                                    align={"left"}
                                                                    variant={"h6"}>{course.title}:
                                                                </Typography>
                                                            </Grid>
                                                            <Grid item item xs={8} xl={10}>
                                                                <Typography align={"left"} variant={'p'}>
                                                                    {courseTopics}
                                                                </Typography>
                                                            </Grid>
                                                        </Grid>
                                                    )
                                                } else return null;
                                            })}
                                        </Grid>}
                                        <Grid item xs={6} xl={6}>
                                            {selectedTopics && selectedTopics.length > 0 &&
                                            <>
                                                {state.student_loading.topics ? <CircularProgress /> :
                                                    <Button
                                                        variant={'contained'}
                                                        onClick={handleGenerateTopicComments}
                                                        disabled={state.student_loading.topics}
                                                    >Generate Topic Comments</Button>
                                                }
                                            </>
                                            }
                                        </Grid>
                                        <Grid item xs={6} xl={6}>
                                            {selectedHighCompetencies.length > 0 &&
                                            <Button variant={'contained'} onClick={handleGenerateCompetencyComments}>Generate competency
                                                comments</Button>
                                            }
                                        </Grid>
                                        <Grid item xs={12} xl={12}>
                                            <form className={classes.root} noValidate autoComplete="off">
                                        <textarea
                                            style={{minWidth: '90%', margin: '20px 50px 20px 50px'}}
                                            aria-label="minimum height"
                                            rows={20}
                                            onChange={handleReportChange}
                                            value={reportText || ''}/>
                                            </form>
                                        </Grid>
                                    </Grid>
                                </Paper>
                            </Grid>
                        </>
                        }
                    </Grid>
                </div>
            </Slide>
        </>
    )
}

