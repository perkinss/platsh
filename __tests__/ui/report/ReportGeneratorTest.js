import React, { useState } from 'react'
import { createRender, createShallow } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import { ReportsContext } from "../../../app/javascript/context/ReportsContext";
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'
import ReportGenerator from "../../../app/javascript/components/report/ReportGenerator";
import {MemoryRouter} from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe('<ReportGenerator />', () => {
    let render, shallow;
    let section, student, state;
    let classes = {
        content: {}
    }

    beforeEach(() => {
        shallow = createShallow();
        render = createRender();
        section = {
            id: 1,
            name: 'Foo'
        }
        student = {
            id: 850,
            name: "Jane",
            unique_id: "Jane",
            school: 2,
            deletable: true,
            grade: "12",
            has_enrollments: false,
            email: "p.sh@w.com",
            pronoun_1: "She",
            pronoun_2: "Her"
        }
        state = {
            topic_averages:
                {
                    1: {
                        850: [{
                            id: 5,
                            title: "Pre-calculus 11",
                            course_marks: [
                                {
                                    id: 13,
                                    name: "Rational Functions",
                                    mark: 83.33333333333333,
                                    weight: 1,
                                },
                                {
                                    id: 15,
                                    name: "Inequalities",
                                    mark: 80,
                                },
                                {
                                    id: 12,
                                    name: "Polynomial Factoring",
                                    mark: 75,
                                }
                            ]
                        },],
                    },
                },
            reporting_periods: [
                {
                    id: 1,
                    name: 'werowrw',
                    contents: [
                        {
                            course_id: 7,
                            id: 111,
                            name: "Applications of Integration",
                        },
                        {
                            course_id: 7,
                            id: 105,
                            name: "Differentiation: Definition and Fundamental Properties",
                        },
                        {
                            course_id: 7,
                            id: 3,
                            name: "Applications of Integration",
                        },
                    ],
                },
                {
                    id: 2,
                    name: 'BAH',
                    contents: [
                        {
                            course_id: 7,
                            id: 1,
                            name: "THOUGHTS of Integration",
                        },
                        {
                            course_id: 7,
                            id: 5,
                            name: "Differentiation: ELUSIVE Properties",
                        },
                        {
                            course_id: 7,
                            id: 11,
                            name: "Applications of PROPERTIES",
                        },
                    ],
                },
            ]
        }
    });

    it('should match the snapshot when hidden', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ReportsContext.Provider value={{dispatch: () => {}, state: state }} >
                    <ReportGenerator classes={classes} showGenerator={false} />
                </ReportsContext.Provider>
            </ThemeProvider>
                );
        expect(wrapper.html()).toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should match the snapshot when loading', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ReportsContext.Provider value={{dispatch: () => {}, state: {} }} >
                    <ReportGenerator classes={classes} student={student} section={section} showGenerator={true} />
                </ReportsContext.Provider>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toMatchSnapshot()
    });

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ReportsContext.Provider value={{state: state }} >
                    <ReportGenerator
                        classes={classes}
                        showGenerator={true}
                        student={student}
                        section={section}
                        testLoadingState={false}
                    />
                </ReportsContext.Provider>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result.toMatchSnapshot)
        expect(result).toContain('Comments for Jane')
        expect(result).toContain('Foo')
        expect(result).not.toContain('<textarea')
    });


})