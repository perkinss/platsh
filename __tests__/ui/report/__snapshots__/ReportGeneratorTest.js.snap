// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportGenerator /> should match the snapshot when hidden 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<ReportGenerator /> should match the snapshot when loading 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-xl-1\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><span class=\\"MuiTypography-root MuiTypography-caption\\">&lt; Back</span></span></button></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-xl-10\\"><h4 class=\\"MuiTypography-root MuiTypography-h4\\">Comments for Jane</h4><h5 class=\\"MuiTypography-root MuiTypography-h5\\">Foo</h5></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-xl-1\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">x</h6></span></button></div></div>"`;
